UMat::UMat(commaInitializer) is declared but has no definition/implementation

Hi all. The codebase declares but does not define/implement the UMat constructor for a `MatCommaInitializer_` comma initializer.

https://github.com/opencv/opencv/blob/049b50d9c0dfb5d06d7528ea334cf42c7bc435f6/modules/core/include/opencv2/core/mat.hpp#L2412

that has no definition. git blame shows that declaration was added 7 years ago, but I can find no matching definition.
Therefore when code using that constructor is compiled, it results in errors like
```
error LNK2001: unresolved external symbol "public: __cdecl cv::UMat::UMat<float>(class cv::MatCommaInitializer_<float> const &)" (??$?0M@UMat@cv@@QEAA@AEBV?$MatCommaInitializer_@M@1@@Z)
fatal error LNK1120: 1 unresolved externals
```

##### System information (version)
- OpenCV => 4.5.0
- Operating System / Platform => Windows 10 64-Bit
- Compiler => Visual Studio Community 2019 v16.8.2

##### Steps to reproduce

Create a UMat with a comma initializer like the following...

```
cv::UMat mymatrix {(cv::Mat_<float>(3, 3) <<
                    1.f, 0.f, 0.f,
                    0.f, 1.f, 0.f,
                    0.f, 0.f, 1.f)};
```

##### Issue submission checklist

 - [X] I report the issue, it's not a question
 - [X] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution
 - [X] I updated to latest OpenCV version and the issue is still there
 - [X] There is reproducer code and related data files: videos, images, onnx, etc

--- a/modules/core/include/opencv2/core/mat.hpp
+++ b/modules/core/include/opencv2/core/mat.hpp
@@ -2432,19 +2432,10 @@ public:
     UMat(const UMat& m, const Rect& roi);
     UMat(const UMat& m, const Range* ranges);
     UMat(const UMat& m, const std::vector<Range>& ranges);
+
+    // FIXIT copyData=false is not implemented, drop this in favor of cv::Mat (OpenCV 5.0)
     //! builds matrix from std::vector with or without copying the data
     template<typename _Tp> explicit UMat(const std::vector<_Tp>& vec, bool copyData=false);
-
-    //! builds matrix from cv::Vec; the data is copied by default
-    template<typename _Tp, int n> explicit UMat(const Vec<_Tp, n>& vec, bool copyData=true);
-    //! builds matrix from cv::Matx; the data is copied by default
-    template<typename _Tp, int m, int n> explicit UMat(const Matx<_Tp, m, n>& mtx, bool copyData=true);
-    //! builds matrix from a 2D point
-    template<typename _Tp> explicit UMat(const Point_<_Tp>& pt, bool copyData=true);
-    //! builds matrix from a 3D point
-    template<typename _Tp> explicit UMat(const Point3_<_Tp>& pt, bool copyData=true);
-    //! builds matrix from comma initializer
-    template<typename _Tp> explicit UMat(const MatCommaInitializer_<_Tp>& commaInitializer);
 
     //! destructor - calls release()
     ~UMat();

